libpromises/processes_select.c

```diff
@@ -1237,29 +1242,44 @@ const char *GetProcessTableLegend(void)
 #if defined(__sun) || defined(TEST_UNIT_TEST)
 static FILE *OpenUcbPsPipe(void)
 {
-    struct stat statbuf;
-    if (stat(UCB_PS, &statbuf) < 0)
+    for (int i = 0; SOLARIS_UCB_STYLE_PS[i]; i++)
     {
-        Log(LOG_LEVEL_VERBOSE,
-            "%s not found, extra process information not available",
-            UCB_PS);
-        return NULL;
-    }
-    if (!(statbuf.st_mode & 0111))
-    {
-        Log(LOG_LEVEL_VERBOSE,
-            "%s not executable, extra process information not available",
-            UCB_PS);
-        return NULL;
-    }
+        struct stat statbuf;
+        if (stat(SOLARIS_UCB_STYLE_PS[i], &statbuf) < 0)
+        {
+            Log(LOG_LEVEL_VERBOSE,
+                "%s not found, cannot be used for extra process information",
+                SOLARIS_UCB_STYLE_PS[i]);
+            continue;
+        }
+        if (!(statbuf.st_mode & 0111))
+        {
+            Log(LOG_LEVEL_VERBOSE,
+                "%s not executable, cannot be used for extra process information",
+                SOLARIS_UCB_STYLE_PS[i]);
+            continue;
+        }
 
-    FILE *cmd = cf_popen(UCB_PS " " UCB_PS_ARGS, "rt", false);
-    if (!cmd)
-    {
-        Log(LOG_LEVEL_WARNING, "Could not execute \"%s %s\", extra process "
-            "information not available.", UCB_PS, UCB_PS_ARGS);
+        char *ps_cmd;
+        xasprintf(&ps_cmd, "%s %s", SOLARIS_UCB_STYLE_PS[i],
+                  SOLARIS_UCB_STYLE_PS_ARGS);
+
+        FILE *cmd = cf_popen(ps_cmd, "rt", false);
+        if (!cmd)
+        {
+            Log(LOG_LEVEL_WARNING, "Could not execute \"%s\", extra process "
+                "information not available.", ps_cmd);
+        }
```

Comment:
It's handled by `cf_pclose`. See that function call inside `LoadPlatformExtraTable()`.